File: ems-frontend/src/services/EmployeeService.js
import axios from 'axios';

const API_BASE_URL = "http://localhost:8080/api/employees";

class EmployeeService {
    deleteEmployee(employeeId) {
        return axios.delete(API_BASE_URL + '/' + employeeId);
    }
    getAllEmployees() {
        return axios.get(API_BASE_URL);
    }

    createEmployee(employee) {
        return axios.post(API_BASE_URL, employee);
    }

    // Add this method to get a single employee by ID
    getEmployeeById(employeeId) {
        return axios.get(API_BASE_URL + '/' + employeeId);
    }

    // Add this method to update an employee
    updateEmployee(employeeId, employee) {
        return axios.put(API_BASE_URL + '/' + employeeId, employee);
    }
}

const employeeService = new EmployeeService();
export default employeeService;

File: ems-backend/src/main/java/com/Deekshith/ems_backend/service/EmployeeService.java
package com.Deekshith.ems_backend.service;

import com.Deekshith.ems_backend.model.Employee;
import java.util.List;

public interface EmployeeService {
    Employee createEmployee(Employee employee);
    Employee getEmployeeById(Long employeeId);
    List<Employee> getAllEmployees();
    Employee updateEmployee(Long employeeId, Employee updatedEmployee);
    void deleteEmployee(Long employeeId);
}